src/Proxy/CommonServiceProxy.php

```diff
@@ -0,0 +1,130 @@
+<?php
+
+namespace Yiisoft\Yii\Debug\Proxy;
+
+use Psr\EventDispatcher\EventDispatcherInterface;
+use Yiisoft\Proxy\ObjectProxy;
+use Yiisoft\Yii\Debug\Collector\CommonServiceCollectorInterface;
+use Yiisoft\Yii\Debug\Event\ProxyMethodCallEvent;
+
+class CommonServiceProxy extends ObjectProxy
+{
+    const LOG_ARGUMENTS = 1 << 0;
+
+    const LOG_RESULT = 1 << 1;
+
+    const LOG_ERROR = 1 << 2;
+
+    private string $service;
+
+    private $logLevel = 0;
+
+    private ?CommonServiceCollectorInterface $collector = null;
+
+    private ?EventDispatcherInterface $dispatcher = null;
+
+    public function __construct(
+        string $service,
+        object $instance,
+        CommonServiceCollectorInterface $collector = null,
+        EventDispatcherInterface $dispatcher = null,
+        int $logLevel = 0
+    ) {
+        $this->service = $service;
+        $this->collector = $collector;
+        $this->dispatcher = $dispatcher;
+        $this->logLevel = $logLevel;
+        parent::__construct($instance);
+    }
+
+    protected function executeMethodProxy(string $methodName, array $arguments, $result, float $timeStart)
+    {
+        $this->log($methodName, $arguments, $result, $timeStart);
+        return $result;
+    }
+
+    protected function getNewStaticInstance(object $instance): ObjectProxy
+    {
+        return new static($this->service, $instance, $this->collector, $this->dispatcher, $this->logLevel);
+    }
+
+    protected function log(string $method, array $arguments, $result, float $timeStart): void
+    {
+        $error = $this->getCurrentError();
+        $this->processLogData($arguments, $result, $error);
+
+        if ($this->collector !== null) {
```

Comment:
Let's move this condition to next private method or remove private method

Reply:
It doesn't make sense.

Reply:
I want to say that need to remove this condition from here and transfer it to method where `$this->collector->collect()` is used

Reply:
> I want to say that need to remove this condition from here and transfer it to method where `$this->collector->collect()` is used

I understand/ That leads to the excess function call.

Reply:
Then private method potentionally has error (call method from `null`)